### PR TITLE
Switching to app based routing

### PR DESCRIPTION
**Pull request by davidfunnell**
-Switched from Page base routing to App based routing with Nextjs13
-Created expected pages in app directory
-Created Navbar and added Links to each of the created pages
-Added "isActive" to link to show currently active page
-Created config/SiteConfig file for main navbar items
-Added types/index.ts file with a type for main nav bar items
-Added personal picture of west for header image to /public

### CODE REVIEW

File: src/app/contact/page.tsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { FC } from 'react'
+
+interface pageProps {
```
Comment (psan2) [inline src/app/contact/page.tsx]: Nitpick - TS types are PascalCase by convention, so this would be PageProps.

I’d also encourage giving you types a specific name even though in this scope it’s the only PageProps around - when debugging, seeing ContactPageProps is much more informative than PageProps. We also don’t need to be worried about optimizing for size, because types get removed at compile time and will never see prod.

File: src/app/contact/page.tsx [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+}
+
+const page: FC<pageProps> = ({}) => {
```
Comment (psan2) [inline src/app/contact/page.tsx]: Same with naming specificity here - React components are PascalCase by convention, and we should be as specific as is helpful. I’d call this ContactPage.

FWIW, standard conventions are:

React components - PascalCase
TS interfaces and types - PascalCase
JS/TS functions or variables - camelCase
TS enums - WHATEVER_THIS_IS

I’d give the rest of these components a pass with the naming convention and specificity in mind 👍

File: src/app/page.tsx [hunk 3]
```diff
@@ -0,0 +1,3 @@
+export default function Home() {
+  return <div>Home Page</div>;
```
Comment (psan2) [inline src/app/page.tsx:2]: Nitpick - we prefer not to export an anonymous function. It’s just a readability thing, but it’s easier to do this as

const HomePage = () => {};

export default HomePage;

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+      <div className="fixed top-0 inset-x-0 bg-zinc-800 border-b border-zinc-600 py-2">
+        <div className="flex justify-center my-2">
+          <div className="flex gap-4 md:gap-9 text-md font-semibold text-gray-400">
+            {MainNavBar.map((link) => {
+              const isActive = pathname === link.href;
+              return (
+                <Link
+                  className={isActive ? "text-white" : ""}
+                  href={link.href}
+                  key={link.title}
```
Comment (psan2) [inline src/components/Navbar.tsx]: Personal preference, but I’d use href instead of title. In theory a target URL will be slightly more unique? 🤷‍♂️

**Review by psan2 — CHANGES_REQUESTED** [review]
Mostly nitpicks! Functionally looks fine, but policing code style is what we have to do when we don’t have eslint really dialed in 😬 thanks dude!

File: src/config/SiteConfig.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { MainNavBarItems } from "@/types";
+
+export const siteConfig = {
+  MainNavBar: [
```
Comment (davidfunnell) [inline src/config/SiteConfig.ts:4]: Should I update MainNavBar be mainNavBar?

File: src/app/contact/page.tsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+interface ContactPageProps {
+
+}
+
+const ContactPage: FC<ContactPageProps> = ({}) => {
+  return <div>Contact page</div>
+}
+
+export default ContactPage
```
Comment (psan2) [inline src/app/contact/page.tsx:11]: Not a blocker for now, but just noticed this - make sure you get the prettier extension installed in your vscode, and probably change your settings to Format on Save. I'll get a prettierrc file in here later.

tl;dr - it's good to use an autoformatter, and if one person on a project doesn't then you end up in this situation where the next person in this file is going to create a bunch of whitespace changes when their formatter adds semicolons to this whole file.

File: src/config/SiteConfig.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+      title: "Join-Us",
+      href: "/join-us",
+      external: false,
+    },
+    {
+      title: "Contact",
+      href: "/contact",
+      external: false,
+    },
+  ] satisfies MainNavBarItems[],
```
Comment (psan2) [inline src/config/SiteConfig.ts:30]: Fancy new TS4.9 shit! Nice.

**Review by psan2 — APPROVED** [review]
